posix/posix.c

```diff
@@ -201,24 +175,61 @@ int posix_getOpenFile(int fd, open_file_t **f)
 }
 
 
+static int _posix_allocfd(process_info_t *p, int fd)
+{
+	fildes_t *nfds;
+	int nfdsz;
+
+	for (; fd < p->maxfd; ++fd) {
+		if (fd >= p->fdsz) {
+			nfdsz = 2 * p->fdsz;
+			if (nfdsz > p->maxfd) {
+				nfdsz = p->maxfd;
+			}
+
+			nfds = vm_kmalloc(nfdsz * sizeof(*nfds));
+			if (nfds == NULL) {
+				return -1;
+			}
+
+			hal_memcpy(nfds, p->fds, p->fdsz * sizeof(*nfds));
+			hal_memset(nfds + p->fdsz, 0, (nfdsz - p->fdsz) * sizeof(*nfds));
+
+			vm_kfree(p->fds);
+
+			p->fds = nfds;
+			p->fdsz = nfdsz;
+		}
+
+		if (p->fds[fd].file == NULL) {
+			return fd;
+		}
```

Comment:
I see a problem in this function.
Such a case:
`fd == 128`, `fdsz == 32`, `maxfd == 1024`
`fd < maxfd` and `fd > fdsz`
you are realocating and then `fdsz == 64`
and in the condition
`p->fds[fd].file == NULL`
array index is out of range.

Can such case occur?
if so, then solution is to change calculation of the value `nfdsz`.

Reply:
Ooops, big bug, thanks for spotting that